lib/wikipedia/client.rb

```diff
@@ -106,7 +108,7 @@ def urlify_value( val )
     def encode( val )
       case val
       when String
-        URI.encode( val, /#{URI::UNSAFE}|[\+&]/ )
+        CGI.escape(val)
```

Comment:
How do you feel about making this `Addressable::URI.encode`? Seems like `CGI.escape` isn't a drop-in replacement, and it's quite possible that this could cause regressions. The test suite for this library is not very comprehensive. But if think this approach is likely safe, I'm not opposed.

Reply:
Using `Addressable::URI.encode` was my first choice, but tests fail with it. Not sure if it means the tests are too specific or other reason.

Reply:
I took a look at this, comparing `URI.encode`, `CGI.escape`, and `Addressable::URI.encode`. The code as written didn't really work well with `CGI.escape`, as that would encode options that are put in the base URL as well as the query parameters.

I just pushed a commit to your branch, rewriting the way this logic works (https://github.com/kenpratt/wikipedia-client/pull/89/commits/7aa230ae8e9edcbf8d62ea6252c173eedf283829). It uses `Addressable::URI.form_encode` to encode the query params options hash, instead of building the string manually. I think this is a bit simpler. I compared the output of all URLs in the test cases using this and the previous approach -- there are some minor differences, mostly where the previous approach was not encoding all special chars in the RFC, and a few minor differences like using `+` instead of `%20`. I tried them all out against the live Wikipedia API, and both URLs resolved to the same output in all cases. So I think this change should be okay, perhaps even a minor improvement. Thoughts?

Reply:
Excellent. I am hoping that we can merge a PR that is only focused on making it work in Ruby 3.x, and immediately after that, perhaps work on things like dependencies and rubocop (although personally, I do not like rubocop).

I have noticed another thing - CI is still running on the deprecated travis.org - it needs to move at least to travis.com, or perhaps to Github actions.

Reply:
Sure, all in one PR is fine. I was thinking perhaps a PR to just tackle the dependency change / removing `Gemfile.lock` first, and then the Ruby 3 upgrade. But I don't feel very strongly about it. I don't think order really matters very much until we cut a new gem release, as most people using the library would be via a gem, presumably.

RE Travis CI, I noticed that deprecation warning. Do you know much about the tradeoffs of moving to the .com version vs Github actions? I'm not very familiar with either.

Reply:
> Do you know much about the tradeoffs of moving to the .com version vs Github actions?

I was a long time user of travis-ci (both .org and .com) and am now 100% on Github Actions. I love travis, they are the grand-daddy of CI, but Actions is too attractive to avoid. I can help with that (opened a separate issue already).

Reply:
Got it, thanks for the context. Moving to Actions sounds good to me. If you want to give it a shot, please go ahead, otherwise I can make time to try it sometime in the next week.